dev/scripts/src/alluxio.org/check-docs/main.go

```diff
@@ -257,6 +273,93 @@ func getSingleRegexMatch(re *regexp.Regexp, l string) (string, error) {
 	return namedMatch, nil
 }
 
+type Subfile struct {
+	Title string `yaml:"title"`
+	URL   string `yaml:"url"`
+}
+
+type File struct {
+	ButtonTitle string    `yaml:"buttonTitle"`
+	Subfiles    []Subfile `yaml:"subfiles"`
+}
+
+// get url from menuPath with two checks one is the buttonTitle check and the second is the url end checks
+func parseMenuURl(menuPath string) (map[string]struct{}, error) {
+	content, err := ioutil.ReadFile(menuPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file at %v with message : \n %v", menuPath, err)
+	}
+	var ret []File
+	if err := yaml.Unmarshal(content, &ret); err != nil {
+		return nil, fmt.Errorf("error unmarshalling config from:\n%v with message : \n %v", string(content), err)
+	}
+	menuMap := map[string]struct{}{}
+	for _, file := range ret {
+		// if buttonTitle have whitespace, the button for list-nav-item in html will not expand
+		if strings.ContainsAny(file.ButtonTitle, " \t\n\r") {
+			return nil, fmt.Errorf("whitespace is not allow in buttonTitle %v, please replace whitespace with _", file.ButtonTitle)
```

Comment:
rather than failing at the first error, collect all the errors as a string and return an error if the list of strings is non-empty
```
var errMsgs []string
...
// if error
errMsgs = append(errMsgs, fmt.Sprintf("error msg %v", arg))
...
if len(errMsgs) > 0 {
  return fmt.Errorf("encountered errors parsing %v:\n%v", menuPath, strings.Join(errMsgs, "\n"))
}
return nil
```